dotnet/src/InternalUtilities/Http/NonDisposableHttpClientHandler.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System;
+using System.Net.Http;
+
+/// <summary>
+/// Represents a singleton implementation of <see cref="HttpClientHandler"/> that is not disposable.
+/// </summary>
+internal sealed class NonDisposableHttpClientHandler : HttpClientHandler
+{
+    /// <summary>
+    /// Lazy-initialized instance of <see cref="NonDisposableHttpClientHandler"/>.
+    /// </summary>
+    private static readonly Lazy<NonDisposableHttpClientHandler> instance = new(() => new NonDisposableHttpClientHandler());
```

Comment:
This doesn't need to be Lazy.  It can just be:
```C#
public static NonDisposableHttpClientHandler Instance { get; } = new NonDisposableHttpClientHandler();
```
The only time this type is accessed is when accessing its Instance member, so the runtime will handle appropriately lazily-initializing it before first use.  Further, even if it wasn't lazy, there are very few resources associated with an unused HttpClientHandler, as the expensive connection resources are themselves created lazily when needed.

Reply:
Got it. It was not about optimal resources usage and more about dealing with concurrency - I remember using Lazy was one of the simplest ways to have singleton out of the box without using locks/double ifs, etc. Not sure if it's still relevan and the Lazy class is good candidate for this, though.

Reply:
```C#
public static NonDisposableHttpClientHandler Instance { get; } = new NonDisposableHttpClientHandler();
```
is thread-safe.  It'll be constructed once and all threads will see the same published instance.  The runtime handles all necessary synchronization around static cctor invocation.  And in this case, the type will end up being beforefieldinit such that the runtime can employ additional optimizations.